### PR TITLE
chore: example message changes between standalone and plugin

### LINKED ISSUE #214: Make examples in usage message parameterizable
The examples that are being added require the command name, which is different whether used as standalone (`faas`) or as a plugin (`kn faas`).

Therefor the example message should be parameterized as a template on this value, which also requires to build up the command dynamically when it is known whether it is running in plugin context or not.

### PR DESCRIPTION
**Pull request by shinigambit**
# Changes
- The example message changes depending on the command being executed as standalone or as a plugin for the `kn` executable

/kind enhancement

Fixes #214

### CODE REVIEW

**Review by lance — COMMENTED** [review]
@shinigambit thanks for this! Because all of the individual commands also have similar examples, maybe `func replaceNameInTemplate(name string) string` could become `func replaceNameInTemplate(name, template string) string` so that other commands could be updated to take advantage of it. WDYT?

**Comment by shinigambit** [comment]
Good idea @lance! I'll change it so it can be reused

File: cmd/root.go [hunk 1]
```diff
@@ -27,21 +41,19 @@ var root = &cobra.Command{
...
-kn func deploy --registry <registry/user>
+	Example: replaceNameInTemplate("func", "example"),
+}
 
-# Curl the service with the service URL
-curl $(kn service describe myfunc -o url)
-`,
+func replaceNameInTemplate(name, template string) string {
+	var buffer bytes.Buffer
+	exampleTemplate.ExecuteTemplate(&buffer, template, name)
```
Comment (lance) [inline cmd/root.go:49]: CI is failing here with
> cmd/root.go:49:33: Error return value of `exampleTemplate.ExecuteTemplate` is not checked (errcheck)
	exampleTemplate.ExecuteTemplate(&buffer, template, name)
Can you check for the error condition here?

I suppose `replaceNameInTemplate()` should return `string, err` and I guess that means that `NewRootCmd()` should return `*cobra.Command, err` as well, which ultimately bubbles up to `plugin.go`.  :thinking:

**Review by lance — COMMENTED** [review]
Thanks for the contribution @shinigambit !

/lgtm
/approve

**Pull request merged**